projects/extension/sql/idempotent/019-worker-tracking.sql

```diff
@@ -0,0 +1,40 @@
+CREATE OR REPLACE FUNCTION ai._worker_start(version text, expected_heartbeat_interval interval) RETURNS uuid AS $$
+DECLARE
+    worker_id uuid;
+BEGIN
+    --can add version check here
+    INSERT INTO ai.vectorizer_worker_connection (version, expected_heartbeat_interval) VALUES (version, expected_heartbeat_interval) RETURNING id INTO worker_id;
+    RETURN worker_id;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION ai._worker_heartbeat(worker_id uuid, num_successes_since_last_heartbeat int, num_errors_since_last_heartbeat int, error_message text) RETURNS void AS $$
+BEGIN
+    UPDATE ai.vectorizer_worker_connection SET 
+          last_heartbeat = now() 
```

Comment:
This may need to be `clock_timestamp()` instead of `now()`. I think this will be called within the transaction that is doing the processing? If so, `now()` will return the time of the start of the transaction every you call this.

Reply:
no, heartbeats are sent via their own connection and txn. but clock_timestamp still probably better